zero/src/bin/aggregator/stdio.rs

```diff
@@ -0,0 +1,112 @@
+use std::io::Read;
+use std::sync::Arc;
+
+use anyhow::{ensure, Result};
+use paladin::runtime::Runtime;
+use tracing::{error, info};
+use zero::{
+    fs::write_proof_to_dir,
+    proof_types::{AggregatableBlockProof, GeneratedBlockProof},
+    prover::ProverConfig,
+};
+
+// TODO(Robin): This should probably live in `leader` binary
+/// Wrapping function for the stdio mode.
+pub(crate) async fn stdio_wrap(
+    runtime: Arc<Runtime>,
+    prover_config: Arc<ProverConfig>,
+) -> Result<()> {
+    let mut buffer = String::new();
+    std::io::stdin().read_to_string(&mut buffer)?;
+
+    let des = &mut serde_json::Deserializer::from_str(&buffer);
+    let block_proof = serde_path_to_error::deserialize::<_, Vec<GeneratedBlockProof>>(des)?
+        .into_iter()
+        .collect::<Vec<_>>();
+
+    ensure!(
+        block_proof.len() == 1,
+        "Expected one and only one block proof to be wrapped, got {:?}",
```

Comment:
`ensure` will already print out the number that fails the condition - you don't need to print it yourself :)